### ISSUE TITLE
Payload Not Sending salesInvoiceLines with Free Text Line

### ISSUE THREAD

**Issue by erimicel** [Sage/SData-2.0#156]
Hi,

Sage 50cloud Accounts (UK) - v26

I have been trying to add M "Message Line" to salesinvoicelines as a new line with given text with payload which looks like this 'shorter version of working one with commidity': (Ruby Nokigiri XML);

```
xml.entry('xmlns:sdata' => 'http://schemas.sage.com/sdata/2008/1',
  'xmlns:crm' => 'http://schemas.sage.com/crmErp/2008',
  'xmlns:sc' => 'http://schemas.sage.com/sc/2009',
  'xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance',
  'xmlns' => 'http://www.w3.org/2005/Atom') {
  xml["sdata"].payload {
     xml["crm"].salesInvoice {
        xml["crm"].active		true
	xml["crm"].type_		"Product Invoice"
        xml["crm"].notes		"Test Note"
        xml["crm"].salesInvoiceLines {
           xml["crm"].salesInvoiceLine {
		xml["crm"].type   "Commentary"
		xml["crm"].text    "TEST Message Line"
           }
        }
    }
}
```

I can add line as commidity with uuid and it works fine with text. but when i try Commentary with free text it creates empty 'M' Message line with 0 quantity 0 price but no given text.

also i have tried to add note with ["crm"].notes "Test Note" but it is goes as empty field as-well.

Just i want to create somehow some side notes to add to invoice layout. Because we can not edit comment lines on invoice lines with sdata.

Thank you

**Comment by darroncockram** [comment]
Looks OK at first glance. You are setting the same properties as I do in the .NET sample I created -
https://github.com/darroncockram/sdatasamples/blob/master/CreateSalesInvoice/Program.cs (lines 101 to 105).

If you capture the resulting XML from your code what does it look like?

**Comment by erimicel** [comment]
i am sending below payload;
```
         xml.entry('xmlns:sdata' => 'http://schemas.sage.com/sdata/2008/1',
		'xmlns:crm' => 'http://schemas.sage.com/crmErp/2008',
		'xmlns:sc' => 'http://schemas.sage.com/sc/2009',
		'xmlns:xsi' => 'http://www.w3.org/2001/XMLSchema-instance',
		'xmlns' => 'http://www.w3.org/2005/Atom') {
		xml["sdata"].payload {
			xml["crm"].salesInvoice {
				xml["crm"].active				true
				xml["crm"].type_				"Product Invoice"
				xml["crm"].text1				"Test1"
				xml["crm"].text2				"Test2"
				xml["crm"].tradingAccount('sdata:uuid' => '1b51a805-0f02-4584-b03d-02fdcb35eab3')
				xml["crm"].date				Date.today.strftime('%Y-%m-%d')
				xml["crm"].taxDate				Date.today.strftime('%Y-%m-%d')
				xml["crm"].customerReference	"0001"
				xml["crm"].salesInvoiceLines {
					xml["crm"].salesInvoiceLine {
						xml["crm"].commodity('sdata:uuid' => "5ba87c24-fa18-4134-9bfb-65a4716c2a8b")
						xml["crm"].text   		"Product Description"
						xml["crm"].quantity	20
						xml["crm"].actualPrice	40
						xml["crm"].netTotal	800
						xml["crm"].grossTotal	800
					}
					xml["crm"].salesInvoiceLine {
						xml["crm"].type		"Commentary"
						xml["crm"].text		"Test Message Line"
					}
				}
			}
		}
	}
```

and,
result XML shorter version returning: 200;

  <published>2020-06-15T18:42:03.000+01:00</published>
  <updated>2020-06-15T18:42:03.000+01:00</updated>
  <http:httpStatus>200</http:httpStatus>
  <sdata:payload>
    <crm:salesInvoice sdata:url="http://erym:5495/sdata/accounts50/GCRM/{C0E8562F-2D62-48C8-9151-48224F9BCB0F}/salesInvoices(427e9665-b6db-41b8-841f-9392beaa37e0)?format=atomentry" sdata:uuid="427e9665-b6db-41b8-841f-9392beaa37e0">
      <crm:active>true</crm:active>
      <crm:reference>5</crm:reference>
      <crm:reference2/>
      <crm:status>Written</crm:status>
      <crm:type>Product Invoice</crm:type>
      <crm:tradingAccount xsi:nil="true"/>
      <crm:customerReference>0001</crm:customerReference>
      <crm:date>2020-06-15</crm:date>
      <crm:time xsi:nil="true"/>
      <crm:taxDate>2020-06-15</crm:taxDate>
      <crm:taxCodes xsi:nil="true"/>
      <crm:currency>GBP</crm:currency>
      <crm:salesInvoiceLines xsi:nil="true"/>
      <crm:lineCount>0</crm:lineCount>
      <crm:netTotal>800.00</crm:netTotal>
      <crm:discountTotal xsi:nil="true"/>
      <crm:chargesTotal xsi:nil="true"/>
      <crm:taxTotal>0.00</crm:taxTotal>
      <crm:grossTotal>800.00</crm:grossTotal>
      <crm:costTotal xsi:nil="true"/>
      <crm:profitTotal xsi:nil="true"/>
      <crm:text1>Test1</crm:text1>
      <crm:text2>Test2</crm:text2>
      <crm:notes xsi:nil="true"/>
      <crm:CustomerId>TEST</crm:CustomerId>
    </crm:salesInvoice>
  </sdata:payload>
</entry>

which on sage invoice created like this;
![Capture](https://user-images.githubusercontent.com/17678162/84689298-b18c1b00-af38-11ea-875a-7fbcec56e6bd.PNG)

just i've mentioned when sending payload SalesInvoiceLine.text carrying all other informations but not text
